### ISSUE TITLE
Travis builds fail on R 3.1 due to vdiffr

### ISSUE THREAD

**Issue by yutannihilation** [tidyverse/ggplot2#2956]
Since the dependency `freetypeharfbuzz` now requires R >= 3.2.

```
Installing package into ‘/home/travis/R/Library’
(as ‘lib’ is unspecified)
ERROR: this R is version 3.1.3, package 'freetypeharfbuzz' requires R  >= 3.2
Error in i.p(...) : 
  (converted from warning) installation of package ‘/tmp/RtmpYMEk9K/file5ad3a366c8a/freetypeharfbuzz_0.2.5.tar.gz’ had non-zero exit status
Calls: <Anonymous> ... with_rprofile_user -> with_envvar -> force -> force -> i.p
Execution halted
The command "Rscript -e 'devtools::install_github(c("lionel-/vdiffr"), build_vignettes = FALSE)'" failed and exited with 1 during .
Your build has been stopped.
```
(https://travis-ci.org/yutannihilation/ggplot2/jobs/444899441#L3542-L3548)

**Comment by clauswilke** [comment]
The problem is this line, which installs the development version of vdiffr:
https://github.com/tidyverse/ggplot2/blob/35507725d82d1d9d91fe5d2e4943671b568b59b2/.travis.yml#L20
Ideally, that install would be executed only when `USE_VDIFFR=true`. Not sure if there's an easy way to do this in Travis.

**Comment by yutannihilation** [comment]
Maybe we can move `r_github_packages` to `r: release` item of the matrix? I'll try later.

**Comment by yutannihilation** [comment]
> Maybe we can move `r_github_packages` to `r: release` item of the matrix?

This seems to fix the problem.

```diff
diff --git a/.travis.yml b/.travis.yml
index 1b6988f3..d4f02e00 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -9,6 +9,7 @@ matrix:
   - r: devel
   - r: release
     env: USE_VDIFFR=true
+    r_github_packages: lionel-/vdiffr
   - r: oldrel
   - r: 3.2
   - r: 3.1
@@ -16,9 +17,6 @@ matrix:
 # work around temporary travis + R 3.5 bug
 r_packages: devtools

-# Install manually because dev vdiffr is not compatible with Appveyor yet
-r_github_packages: lionel-/vdiffr
-
 # environment variables set for all builds
 env:
   global:
```
(https://github.com/yutannihilation/ggplot2/compare/master...e01dff5ce9c89d9a794ca41dca4f6e8f30b4fe5a)

But, I see different errors on R 3.2 and R 3.1 due to the new devtools' warning, which `warnings_are_errors: false` won't help at this stage...

```r
Skipping 2 packages not available: rgeos, sf
Installing 77 packages: ...snip...
Installing packages into ‘/home/travis/R/Library’
(as ‘lib’ is unspecified)
Error: (converted from warning) packages ‘rgeos’, ‘sf’ are not available (for R version 3.1.3)
Execution halted
```
